tests/unit/commands/samconfig/test_samconfig.py

```diff
@@ -162,6 +162,65 @@ def test_build(self, do_cli_mock):
                 "READ",
             )
 
+    @patch("samcli.commands.build.command.do_cli")
+    def test_build_with_no_use_container(self, do_cli_mock):
+        config_values = {
+            "resource_logical_id": "foo",
+            "template_file": "mytemplate.yaml",
+            "base_dir": "basedir",
+            "build_dir": "builddir",
+            "cache_dir": "cachedir",
+            "cache": True,
+            "use_container": False,
+            "manifest": "requirements.txt",
+            "docker_network": "mynetwork",
+            "skip_pull_image": True,
+            "parameter_overrides": "ParameterKey=Key,ParameterValue=Value ParameterKey=Key2,ParameterValue=Value2",
+            "container_env_var": [("")],
+            "container_env_var_file": "file",
+            "build_image": [("")],
+            "exclude": [("")],
+            "mount_with": "read",
+        }
+
+        with samconfig_parameters(["build"], self.scratch_dir, **config_values) as config_path:
+            from samcli.commands.build.command import cli
+
+            LOG.debug(Path(config_path).read_text())
+            runner = CliRunner()
+            result = runner.invoke(cli, [])
```

Comment:
What you probably want to check here is that the parameter in the **command line** gets picked up (`--no-use-container`, that you're adding). To test that, you need to pass it in this part. Like [this line](https://github.com/aws/aws-sam-cli/blob/36e033b8d01e4ea13844023b9e3070f108389b67/tests/unit/commands/samconfig/test_samconfig.py#L251). 

Otherwise, you're just testing that the samconfig works fine, not the command line argument. (Probably the same for sync below)

Reply:
Both tests are different. The use cases in the link you provided focus on testing overrides. For example, if my configuration includes `"use_container": True`, I can pass `--no-use-container` to override it, effectively setting `"use_container": False`.

This is a draft PR to validate whether make schema works. Please note that the PR is not complete as you review it.

Reply:
I suppose we could have 1 unit test for testing the default config value as here and another one like @valerena mentioned about with `use-container=True` and override it with `--no-use-container` to assert it gets set to `False`. The `sam sync` test below could be updated with the latter case since both cases stem from the same click option anyways (we might not need duplicate tests for each command).